File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        
        for(std::vector<int>::iterator it = nums.begin(); it != nums.end(); ++it){
            int num = target - *it;
            int distance = it - nums.begin();
            if(umap.find(num) == umap.end()){
                 cout <<" not found\n\n";
            }else{
                return vector<int>{distance, umap.find(num)->second};
            }
            
            umap[*it] = distance;
            
            
        }
        

        
        return vector<int>(0);
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            dict[nums[i]] = i
        list = []
        for i in range(len(nums)):
            diff = target-nums[i]
            if diff in dict and dict[diff]!= i:
                list.append(dict[diff]) 
                list.append(i)
                return list
        return []
        
        